src/fabric.c

```diff
@@ -624,7 +624,7 @@ static void ofi_ini_dir(const char *dir)
 	char *lib;
 	struct dirent **liblist = NULL;
 
-	n = scandir(dir, &liblist, lib_filter, NULL);
+	n = scandir(dir, &liblist, lib_filter, alphasort);
```

Comment:
Can you elaborate how "providers being loaded can be controlled by tweaking the file names"?

Reply:
scandir(), if no comp_func supplied, returns the entries in storage order, which does not relate to the file name. Which the fix, the list will be sorted in alphabeta order.  If we have two versions of the same provider in the same directory, we can control which one is tried first by the file name. for example libverbs-1-fi.so would be tried before libverbs-2-fi.so.